CMakeLists.txt

```diff
@@ -205,6 +207,98 @@ if (NN_ENABLE_NANOCAT)
+        message (STATUS "Using asciidoctor at ${ASCIIDOCTOR_EXE}")
+    endif ()
+endif ()
+
+# Build the documenation
+if (NN_ENABLE_DOC)
+
+    set (NN_DOCDIR ${CMAKE_CURRENT_SOURCE_DIR}/doc)
+    set (NN_STYLESHEET ${NN_DOCDIR}/stylesheet.css)
+    set (NN_TITLE ${PROJECT_NAME} ${NN_PACKAGE_VERSION})
+    set (NN_A2M ${ASCIIDOCTOR_EXE} -b manpage -amanmanual='${NN_TITLE}')
+    set (NN_A2H ${ASCIIDOCTOR_EXE} -d manpage -b html5 -a stylesheeet=${NN_STYLESHEET} -aversion-label=${PROJECT_NAME} -arevnumber=${NN_PACKAGE_VERSION})
+
+    macro (add_libnanomsg_man NAME SECT)
+        add_custom_command (
+            OUTPUT ${NAME}.${SECT}
+            COMMAND ${NN_A2M} -o ${NAME}.${SECT} ${NN_DOCDIR}/${NAME}.txt
+            MAIN_DEPENDENCY ${NN_DOCDIR}/${NAME}.txt
+        )
+
+        add_custom_command (
+            OUTPUT ${NAME}.${SECT}.html
+            COMMAND ${NN_A2H} -o ${NAME}.${SECT}.html ${NN_DOCDIR}/${NAME}.txt
+            DEPENDS ${NN_STYLESHEET}
+            MAIN_DEPENDENCY ${NN_DOCDIR}/${NAME}.txt
+        )
+
+        set(NN_MANS ${NN_MANS} ${NAME}.${SECT})
+        set(NN_HTMLS ${NN_HTMLS} ${NAME}.${SECT}.html)
+
+        install (
+            FILES ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.${SECT}.html
+            DESTINATION ${CMAKE_INSTALL_DOCDIR}
+        )
+        install (
+            FILES ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.${SECT}
+            DESTINATION ${CMAKE_INSTALL_MANDIR}/man${SECT}
+        )
+
+    endmacro (add_libnanomsg_man)
+
+    if (NN_ENABLE_NANOCAT)
+        add_libnanomsg_man (nanocat 1)
+    endif ()
+
+    add_libnanomsg_man (nn_errno 3)
+    add_libnanomsg_man (nn_strerror 3)
+    add_libnanomsg_man (nn_symbol 3)
+    add_libnanomsg_man (nn_symbol_info 3)
+    add_libnanomsg_man (nn_allocmsg 3)
+    add_libnanomsg_man (nn_reallocmsg 3)
+    add_libnanomsg_man (nn_freemsg 3)
+    add_libnanomsg_man (nn_socket 3)
+    add_libnanomsg_man (nn_close 3)
+    add_libnanomsg_man (nn_get_statistic 3)
+    add_libnanomsg_man (nn_getsockopt 3)
+    add_libnanomsg_man (nn_setsockopt 3)
+    add_libnanomsg_man (nn_bind 3)
+    add_libnanomsg_man (nn_connect 3)
+    add_libnanomsg_man (nn_shutdown 3)
+    add_libnanomsg_man (nn_send 3)
+    add_libnanomsg_man (nn_recv 3)
+    add_libnanomsg_man (nn_sendmsg 3)
+    add_libnanomsg_man (nn_recvmsg 3)
+    add_libnanomsg_man (nn_device 3)
+    add_libnanomsg_man (nn_cmsg 3)
+    add_libnanomsg_man (nn_poll 3)
+
+    add_libnanomsg_man (nanomsg 7)
+    add_libnanomsg_man (nn_pair 7)
+    add_libnanomsg_man (nn_reqrep 7)
+    add_libnanomsg_man (nn_pubsub 7)
+    add_libnanomsg_man (nn_survey 7)
+    add_libnanomsg_man (nn_pipeline 7)
+    add_libnanomsg_man (nn_bus 7)
+    add_libnanomsg_man (nn_inproc 7)
+    add_libnanomsg_man (nn_ipc 7)
+    add_libnanomsg_man (nn_tcp 7)
+    add_libnanomsg_man (nn_ws 7)
+    add_libnanomsg_man (nn_env 7)
+
+     add_custom_target (man ALL DEPENDS ${NN_MANS})
+     add_custom_target (html ALL DEPENDS ${NN_HTMLS})
```

Comment:
No its real.  I fixed it.  I *hate hate hate* space based indentation.  8-cell tabs (with real tab characters) are *soo* much easier to see and use.  (This is a topic we may explore later.)